src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -4231,7 +4231,7 @@ public bool ConnectAsync(SocketAsyncEventArgs e)
                     {
                         InternalBind(new IPEndPoint(IPAddress.Any, 0));
                     }
-                    else
+                    else if (endPointSnapshot.AddressFamily == AddressFamily.InterNetworkV6)
```

Comment:
> So, are you saying that Desktop-based implementation must be also fixed to check for the address family here? But that's a different repo, and it must be a different work item? I can open a bug there, what repo is this?

@SidharthNabar PTAL

I presume that your code won't work on Windows anyway? If so we may be OK with only changing the corefx repo and ensuring that the rest of the behaviors are in line with Desktop (e.g. the test for Windows fails the same way on N and Desktop platforms).